### DISCUSSION TITLE
Is there a way to check whether the cili has already been downloaded?

### DISCUSSION THREAD

**Discussion by francis-dion** [goodmami/wn#172] [Q&A] [answered]
I'm about to deploy my solution and I don't want it to download files already downloaded every time it starts.
For lexicons, it can check wn.lexicons() to see if the ones it needs are already present.
As per the documentation, "CILI can be downloaded just like a lexicon"
`wn.download('cili:1.0')`
Yet, 'cili' doesn't showup in wn.lexicons().
Am I missing something? Any suggestions?
Thanks!

**Comment by francis-dion** [comment, marked as answer]
Nevermind, I found the answer!
All I had to do is check the wn.projects().
`if 'cili' not in [proj['id'] for proj in wn.projects()]:`

**Comment by goodmami** [comment]
@francis-dion that is actually not an answer. `wn.projects()` checks Wn's [index](https://github.com/goodmami/wn/blob/main/wn/index.toml), which is a list of projects Wn knows about so it can download them by the project name instead of the URL. It does not say whether the project has been added to the database, but the `cache` value will be filled if it has been downloaded.

You've discovered a missing feature of Wn, but for now this might work:

```pycon
>>> import wn
>>> wn.ili('i1').status
'active'
```

The status of an ILI is `active` if CILI has been added to the database. If not, the status will be `presupposed` if the ILI has been observed in a lexicon.

Also apparently missing is the ability to remove CILI from the database:

```pycon
>>> wn.download('cili')
[...]
>>> wn.remove('cili')
[...]
wn.Error: no lexicon found with lang=None and lexicon='cili'
```
